pyanaconda/core/util.py

```diff
@@ -165,6 +167,68 @@ def preexec():
     return partsubp(preexec_fn=preexec)
 
 
+class WaylandStatus:
+    """Status of Wayland launch.
+
+    Values of an instance can be modified from the handler functions.
+    """
+    def __init__(self):
+        self.started = False
+        self.timed_out = False
+
+    def needs_waiting(self):
+        return not (self.started or self.timed_out)
+
+
+def startWl(weston_config=WESTON_CONFIG, output_redirect=None, timeout=WAYLAND_TIMEOUT):
+    """ Start Weston for Wayland and return once Weston is ready to accept connections.
+
+        We can identify whether Weston is ready by testing if
+        the Wayland socket is open yet. Once it is, we can return success.
+
+        :param weston_config: The weston.ini(5) configuration to use, as a dictionary
+        :param output_redirect: file or file descriptor to redirect stdout and stderr to
+        :param timeout: Number of seconds to timing out.
+    """
+    wl_status = WaylandStatus()
+
+    # Create the config file for Weston
+    weston_config_file = tempfile.NamedTemporaryFile(suffix="-wl-weston-sysinstall-ini",
+                                                     delete=False)
+    weston_config_ini = configparser.ConfigParser()
+    for section, options in weston_config.items():
+        weston_config_ini.add_section(section)
+        for key, value in options.items():
+            weston_config_ini.set(section, key, str(value))
+    weston_config_ini.write(weston_config_file)
+
+    log.debug("Starting Weston.")
+    argv = ["weston", f"--config={weston_config_file}", "--log=/tmp/weston.log",
+            f"--socket={WAYLAND_SOCKET_NAME}"]
+
+    childproc = startProgram(argv, stdout=output_redirect, stderr=output_redirect)
+    WatchProcesses.watch_process(childproc, argv[0])
+
+    for _ in range(0, timeout):
+        try:
+            xdg_runtime_dir = os.getenv("XDG_RUNTIME_DIR")
+            path = os.path.join(xdg_runtime_dir, WAYLAND_SOCKET_NAME)
+            sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM, 0)
+            sock.connect(path)
+            sock.close()
+            wl_status.started = True
+            return wl_status.started
+        except Exception:
```

Comment:
In theory, I could, but I have intentionally not done so because I want it to happen in any exception case. It's merely for shorting the loop to iterate on the timeout.